File: common/src/main/java/org/sandboxpowered/silica/network/play/clientbound/UpdateChunkPosition.java
package org.sandboxpowered.silica.network.play.clientbound;

import org.sandboxpowered.silica.network.PacketByteBuf;
import org.sandboxpowered.silica.network.PacketHandler;
import org.sandboxpowered.silica.network.PacketPlay;
import org.sandboxpowered.silica.network.PlayContext;

public class UpdateChunkPosition implements PacketPlay {
    private int x, y;

    public UpdateChunkPosition() {
    }

    public UpdateChunkPosition(int x, int y) {
        this.x = x;
        this.y = y;
    }

    @Override
    public void read(PacketByteBuf buf) {

    }

    @Override
    public void write(PacketByteBuf buf) {
        buf.writeVarInt(x);
        buf.writeVarInt(y);
    }

    @Override
    public void handle(PacketHandler packetHandler, PlayContext context) {

    }
}

File: common/src/main/kotlin/org/sandboxpowered/silica/network/play/clientbound/UpdateChunkPosition.kt
package org.sandboxpowered.silica.network.play.clientbound

import org.sandboxpowered.silica.network.PacketByteBuf
import org.sandboxpowered.silica.network.PacketHandler
import org.sandboxpowered.silica.network.PacketPlay
import org.sandboxpowered.silica.network.PlayContext

class UpdateChunkPosition(private var x: Int = 0, private var y: Int = 0) : PacketPlay {

    override fun read(buf: PacketByteBuf) {}
    override fun write(buf: PacketByteBuf) {
        buf.writeVarInt(x)
        buf.writeVarInt(y)
    }

    override fun handle(packetHandler: PacketHandler, context: PlayContext) {}
}